chunks/chunks.go

```diff
@@ -345,6 +345,15 @@ func (s *Reader) Close() error {
 	return closeAll(s.cs...)
 }
 
+// Size returns the size of the chunks.
+func (s *Reader) Size() int64 {
```

Comment:
Instead of calculating this everytime we call `Size()`, it is possible to calculate only once and store it in `Reader`? I believe this will optimise calculation of `prometheus_tsdb_storage_blocks_bytes_total` metric.

Reply:
yes, good suggestion, thanks, I have refactored it.